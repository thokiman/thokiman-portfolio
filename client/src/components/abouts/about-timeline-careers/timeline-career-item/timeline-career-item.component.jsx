import React from 'react';
import {
  TimelineCareerCompany,
  TimelineCareerDateLength,
  TimelineCareerDescription,
  TimelineCareerLocation,
  TimelineCareerTitleType,
  TimelineCareerYear,
  TimelineCareerYearElement,
  TimelineTextContainer,
} from './timeline-career-item.styles';

export const TimelineCareerItem = ({ beforeWorking }) => {
  return (
    <TimelineTextContainer>
      {beforeWorking.map(
        ({
          id,
          year,
          company,
          title,
          type,
          date,
          length,
          location,
          description,
        }) => {
          return (
            <TimelineCareerYearElement key={id} name={`${year}`}>
              <TimelineCareerYear>{year}</TimelineCareerYear>
              <TimelineCareerCompany>{company}</TimelineCareerCompany>
              <TimelineCareerTitleType>
                {title}
                <br />
                {type}
              </TimelineCareerTitleType>
              <TimelineCareerDateLength>
                {date}
                <br />
                {length}
              </TimelineCareerDateLength>
              <TimelineCareerLocation>{location}</TimelineCareerLocation>
              <TimelineCareerDescription>
                {description}
              </TimelineCareerDescription>
            </TimelineCareerYearElement>
          );
        }
      )}
    </TimelineTextContainer>
  );
};

export default TimelineCareerItem;
