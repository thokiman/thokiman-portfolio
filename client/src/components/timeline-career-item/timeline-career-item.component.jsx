import React from "react";
import { CSSTransitionGroup } from "react-transition-group";

import {
  TimelineTextContainer,
  TimelineCareerYearElement,
  TimelineCareerYear,
  TimelineCareerCompany,
  TimelineCareerTitleType,
  TimelineCareerDateLength,
  TimelineCareerDescription,
  TimelineCareerLocation,
} from "./timeline-career-item.styles";

const TimelineCareerItem = ({ beforeWorking }) => {
  return (
    <TimelineTextContainer>
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={200}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        {beforeWorking.map(
          ({
            id,
            year,
            company,
            title,
            type,
            date,
            length,
            location,
            description,
          }) => {
            return (
              <TimelineCareerYearElement key={id} name={`${year}`}>
                <TimelineCareerYear>{year}</TimelineCareerYear>
                <TimelineCareerCompany>{company}</TimelineCareerCompany>
                <TimelineCareerTitleType>
                  {title}
                  <br />
                  {type}
                </TimelineCareerTitleType>
                <TimelineCareerDateLength>
                  {date}
                  <br />
                  {length}
                </TimelineCareerDateLength>
                <TimelineCareerLocation>{location}</TimelineCareerLocation>
                <TimelineCareerDescription>
                  {description}
                </TimelineCareerDescription>
              </TimelineCareerYearElement>
            );
          }
        )}
      </CSSTransitionGroup>
    </TimelineTextContainer>
  );
};

export default TimelineCareerItem;
