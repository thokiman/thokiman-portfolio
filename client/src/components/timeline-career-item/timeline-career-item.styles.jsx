import styled, { css } from "styled-components";
import { Element } from "react-scroll";

export const TimelineTextContainer = styled.div`
  font-family: futura-book;
`;
export const TimelineCareerYearElement = styled(Element)`
  margin: 10px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: repeat(5, 1fr);
  gap: 10px 10px;
  grid-template-areas:
    "timeline-career-year timeline-career-year timeline-career-company timeline-career-company"
    "timeline-career-title-type timeline-career-title-type timeline-career-company timeline-career-company"
    "timeline-career-title-type timeline-career-title-type timeline-career-description timeline-career-description"
    "timeline-career-date-length timeline-career-date-length timeline-career-description timeline-career-description"
    "timeline-career-location timeline-career-location timeline-career-description timeline-career-description";
  background-color: rgba(116, 198, 212, 0.5);
`;

const TimelineCareerTextStyles = css`
  color: #bccfd5;
  font-kerning: auto;
  margin-bottom: 10px;
  text-align: left;
`;
export const TimelineCareerYear = styled.div`
  ${TimelineCareerTextStyles}
  grid-area: timeline-career-year;
  text-align: left;
  margin: 15px 0;
  font-size: 350%;
  letter-spacing: 0.5em;
  padding-left: 30px;
  text-shadow: #74c6d4 10px 23px;
  @media only screen and (max-width: 800px) {
    font-size: 250%;
  }
`;

export const TimelineCareerCompany = styled.div`
  ${TimelineCareerTextStyles}
  grid-area: timeline-career-company;
  font-size: 3.6vw;
  line-height: 1.2em;
  letter-spacing: 0.15em;
  padding: 10px 20px;
  position: relative;
  top: 91px;
  @media only screen and (max-width: 800px) {
    font-size: 150%;
    position: relative;
    right: 20px;
  }
  @media only screen and (max-width: 370px) {
    font-size: 120%;
    position: relative;
    right: 20px;
  }
  @media only screen and (max-width: 289px) {
    font-size: 100%;
    position: relative;
    right: 20px;
  }
  @media only screen and (max-width: 244px) {
    font-size: 80%;
    position: relative;
    right: 20px;
  }
`;
export const TimelineCareerTitleType = styled.div`
  ${TimelineCareerTextStyles}
  grid-area: timeline-career-title-type;
  font-family: futura-light;
  font-size: 3vw;
  text-align: right;
  line-height: 1.2em;
  letter-spacing: 0.15em;
  padding: 10px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  @media only screen and (max-width: 800px) {
    font-size: 140%;
    font-weight: bold;
    padding: 4px;
    @media only screen and (max-width: 370px) {
      font-size: 110%;
      font-weight: bold;
      padding: 4px;
    }
    @media only screen and (max-width: 289px) {
      font-size: 80%;
      font-weight: bold;
      padding: 4px;
    }
  }
`;

export const TimelineCareerDateLength = styled.div`
  ${TimelineCareerTextStyles}

  grid-area: timeline-career-date-length;
  font-family: futura-light;
  font-size: 2.4vw;
  line-height: 1.5em;
  text-align: right;
  letter-spacing: 0.15em;
  padding: 10px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  @media only screen and (max-width: 800px) {
    font-size: 100%;
    font-weight: bold;
    padding: 4px;
  }
`;

export const TimelineCareerLocation = styled.div`
  ${TimelineCareerTextStyles}
  grid-area: timeline-career-location;
  font-family: futura-light;
  font-size: 2.4vw;
  text-align: right;
  letter-spacing: 0.1em;
  padding: 10px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  @media only screen and (max-width: 800px) {
    font-size: 100%;
    font-weight: bold;

    padding: 4px;
  }
`;
export const TimelineCareerDescription = styled.div`
  ${TimelineCareerTextStyles}
  grid-area: timeline-career-description;
  font-family: futura-light;
  font-size: 2.3vw;
  letter-spacing: 0.05em;
  display: flex;
  align-items: center;
  justify-content: center;
  line-height: 1.3em;
  margin-top: 60px;
  padding: 10px 20px;
  @media only screen and (max-width: 800px) {
    font-size: 100%;
    letter-spacing: 0.08em;
    font-weight: bold;
    text-align: left;
    padding: 4px;
  }
`;
TimelineTextContainer.displayName = "TimelineTextContainer";
TimelineCareerYearElement.displayName = "TimelineCareerYearElement";
TimelineCareerYear.displayName = "TimelineCareerYear";
TimelineCareerCompany.displayName = "TimelineCareerCompany";
TimelineCareerTitleType.displayName = "TimelineCareerTitleType";
TimelineCareerDateLength.displayName = "TimelineCareerDateLength";
TimelineCareerLocation.displayName = "TimelineCareerLocation";
TimelineCareerDescription.displayName = "TimelineCareerDescription";
