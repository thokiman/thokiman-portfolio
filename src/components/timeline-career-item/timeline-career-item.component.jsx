import React from "react";
import { Element } from "react-scroll";
import { CSSTransitionGroup } from "react-transition-group";

import "./timeline-career-item.styles.scss";

const TimelineCareerItem = ({ beforeWorking }) => {
  return (
    <div className="timeline-text-container">
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={200}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        {beforeWorking.map(
          ({
            id,
            year,
            company,
            title,
            type,
            date,
            length,
            location,
            description,
          }) => {
            return (
              <Element
                key={id}
                name={`${year}`}
                className="timeline-career-year-element"
              >
                <div className="timeline-career-year">{year}</div>
                <div className="timeline-career-company">{company}</div>
                <div className="timeline-career-title-type">
                  {title}
                  <br />
                  {type}
                </div>
                <div className="timeline-career-date-length">
                  {date}
                  <br />
                  {length}
                </div>
                <div className="timeline-career-location">{location}</div>
                <div className="timeline-career-description">{description}</div>
              </Element>
            );
          }
        )}
      </CSSTransitionGroup>
    </div>
  );
};

export default TimelineCareerItem;
