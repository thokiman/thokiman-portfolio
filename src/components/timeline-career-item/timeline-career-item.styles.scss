.timeline-text-container {

    font-family: futura-book;

    .timeline-career-year-element {
        margin: 10px;

        height: 200%;
        display: grid;
        grid-template-columns: repeat(4, 1fr);
        grid-auto-rows: repeat(5, 1fr);
        gap: 10px 10px;
        grid-template-areas:
            "timeline-career-year timeline-career-year timeline-career-company timeline-career-company"
            "timeline-career-title-type timeline-career-title-type timeline-career-company timeline-career-company"
            "timeline-career-title-type timeline-career-title-type timeline-career-description timeline-career-description"
            "timeline-career-date-length timeline-career-date-length timeline-career-description timeline-career-description"
            "timeline-career-location timeline-career-location timeline-career-description timeline-career-description";
        background-color: rgba($color: #128194, $alpha:0.6);

        .timeline-career-year {
            grid-area: timeline-career-year;
            margin-top: 15px;
            font-size: 350%;
            color: #bccfd5;
            letter-spacing: 0.5em;
            font-kerning: auto;
            padding-left: 30px;
            margin-bottom: 10px;
            text-align: left;
            text-shadow: #74c6d4 10px 23px;

        }

        .timeline-career-company {

            grid-area: timeline-career-company;
            font-size: 450%;
            line-height: 1.2em;
            color: #bccfd5;
            text-align: left;
            letter-spacing: 0.15em;
            font-kerning: auto;
            padding: 10px;

            margin-bottom: 10px;
            position: relative;
            top: 91px;
        }

        .timeline-career-title-type {

            grid-area: timeline-career-title-type;
            font-family: futura-light;
            font-size: 275%;
            line-height: 1.2em;
            color: #bccfd5;
            text-align: center;
            letter-spacing: 0.15em;
            font-kerning: auto;
            padding: 10px;
            display: flex;
            align-items: center;
            justify-content: center;
            margin-bottom: 10px;
        }

        .timeline-career-date-length {

            grid-area: timeline-career-date-length;
            font-family: futura-light;
            font-size: 200%;
            color: #bccfd5;
            line-height: 1.5em;
            text-align: center;
            letter-spacing: 0.15em;
            font-kerning: auto;
            padding: 10px;
            display: flex;
            align-items: center;
            justify-content: center;
            margin-bottom: 10px;
        }

        .timeline-career-location {

            grid-area: timeline-career-location;
            font-family: futura-light;
            font-size: 200%;
            color: #bccfd5;
            text-align: center;
            letter-spacing: 0.1em;
            font-kerning: auto;
            padding: 10px;
            display: flex;
            align-items: center;
            justify-content: center;
            margin-bottom: 10px;
        }

        .timeline-career-description {

            grid-area: timeline-career-description;
            font-family: futura-light;
            font-size: 180%;
            color: #bccfd5;
            text-align: left;
            letter-spacing: 0.05em;
            font-kerning: auto;
            display: flex;
            align-items: center;
            justify-content: center;
            line-height: 1.3em;

            margin-top: 60px;
            margin-bottom: 10px;
            padding: 10px;
        }

    }
}