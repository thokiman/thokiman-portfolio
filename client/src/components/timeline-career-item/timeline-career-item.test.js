import React from "react";
import { shallow } from "enzyme";

import { TimelineCareerItem } from "./timeline-career-item.component";

describe("TimelineCareerItem component", () => {
  let wrapper;
  const mockArray = [];
  beforeEach(() => {
    const mockProps = {
      beforeWorking: mockArray,
    };
    wrapper = shallow(<TimelineCareerItem {...mockProps} />);
  });

  it("should render TimelineCareerItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
