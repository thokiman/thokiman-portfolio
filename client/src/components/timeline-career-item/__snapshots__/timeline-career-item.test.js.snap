// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineCareerItem component should render TimelineCareerItem component 1`] = `ShallowWrapper {}`;
